### PR TITLE
Add a wrapper allowing safe evidence collection

### PR DESCRIPTION
**Pull request by kubasobon**
### Summary of your changes

Add a new common function `collect_evidence()` which allows for safe evidence collection, even in cases when some values are `undefined`.

### Screenshot/Data

image

### Related Issues

Related: https://github.com/elastic/cloudbeat/issues/1509

### Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

#### Introducing a new rule?

- [ ] Generate rule metadata using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rules-metadata)
- [ ] Add relevant unit tests
- [ ] Generate relevant rule templates using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rule-templates), and open a PR in [elastic/packages/cloud_security_posture](https://github.com/elastic/integrations/tree/main/packages/cloud_security_posture)

### CODE REVIEW

File: security-policies/bundle/compliance/lib/common/common.rego [hunk 1]
```diff
@@ -33,6 +34,13 @@ calculate_result(evaluation) = "passed" if {
 	evaluation
 } else = "failed"
 
+# Safely evaluate evidence. In case a key is undefined, it will be defaulted.
+# keypaths is an object defined as {str: array}
+collect_evidence(resource, key_paths) := {key: evidence |
+	some key, path in key_paths
+	evidence := object.get(resource, path, "<undefined>")
```
Comment (oren-zohar) [inline security-policies/bundle/compliance/lib/common/common.rego:41]: Are we going to replace the `<undefined>` with full resource instead?

Comment (orouz) [reply security-policies/bundle/compliance/lib/common/common.rego:41]: [seems like](https://github.com/elastic/cloudbeat/issues/1509#issuecomment-1909686173) the FE will display a specific message to indicate the evidence collection failed, in which case sending `<undefined>` would be one way to identify this case. if we send the resource (which is already sent at the top level) the FE wouldn't be able to differentiate when evidence was collected or not.

**Review by orouz — APPROVED** [review]
LGTM

File: security-policies/bundle/compliance/lib/common/common.rego (hunk 1 above)
Comment (oren-zohar) [reply security-policies/bundle/compliance/lib/common/common.rego:41]: @orouz Let's follow [product's call](https://github.com/elastic/cloudbeat/issues/1509#issuecomment-1906789385) on this for now

**Review by oren-zohar — CHANGES_REQUESTED** [review]
Based on [product's request](https://github.com/elastic/cloudbeat/issues/1509#issuecomment-1906789385) let's send the full resource as a fallback instead of `"<undefined>"` for now

**Review by oren-zohar — APPROVED** [review]

**Pull request merged**